src/NLog/LogEventInfo.cs

```diff
@@ -231,12 +240,12 @@ public object[] Parameters
         /// </summary>
         public IFormatProvider FormatProvider
         {
-            get { return formatProvider; }
+            get { return this.logMessageFormatter?.FormatProvider; }
             set
             {
-                if (formatProvider != value)
+                if (logMessageFormatter?.FormatProvider != value)
```

Comment:
got a resharper warning here, not sure if correct

![image](https://user-images.githubusercontent.com/5808377/28186145-55975fc0-681a-11e7-8a9e-1287a0ab3581.png)